src/Microsoft.VisualStudio.Services.Agent/Terminal.cs

```diff
@@ -20,12 +22,33 @@ public interface ITerminal: IAgentService
         void WriteLine();
         void WriteLine(string line);
         void WriteError(string line);
+        void UnregisterCancelEvent();
     }
     
     public sealed class Terminal: AgentService, ITerminal
     {
         public bool Silent { get; set; }
 
+        public event EventHandler CancelKeyPress;
+
+        public override void Initialize(IHostContext hostContext)
+        {
+            base.Initialize(hostContext);
+
+            Console.CancelKeyPress += Console_CancelKeyPress;            
+        }
+
+        public void UnregisterCancelEvent()
```

Comment:
good suggestion. will make it IDisposable and will dispose from the Program.MainAsync